### ISSUE TITLE
console to iou and vpcs has title "Terminal" while it should be the device name in the project

### ISSUE THREAD

**Issue by ahbali** [GNS3/gns3-gui#2480]
for iou and vpcs when opening a console, the %d in  ( gnome-terminal -t "%d" -e "telnet %h %p" ) has value "Terminal" and it should be like "iou1" or "pc-1" ( device name in the project ).
the behaviour is ok with dynamips devices.

using ubuntu 16.04 and gns3 ppa.

**Comment by n3oxmind** [comment]
@ahbali @grossmj 
I got the same problem after installing the `gns3-gui-2.1.5`, and didn't check why, I though its a bug after seeing it already been reported.  BUT
I found out its a terminal issue rather than a gns3 bug.
I have tested gns3 with multiple terminal including gnome-terminal it worked fine except `tilix with muliple tabs(sessions)`

**For gnome-terminal** 
make sure that `When terminal commands set their own titles:` under `Profile Preferences -> Command` is set to `Replace initial title` and not `Keep initial title`
![sw2_015](https://user-images.githubusercontent.com/10103340/39658349-bde0a54c-4fc7-11e8-9c8b-ed81c1bacb5a.png)
**For Tilix terminal**
if you have few telnet sessions < 4 then use this command
```
tilix -a session-add-down -t "%d" -e "telnet %h %p"
```
if you have multiple telnet session > 4 
```
tilix -a app-new-session -t "%d" -e "telnet %h %p"
```
and make sure to set these Tilix variables: Default session name to `${title}` and Application title to `${activeTerminalTitle}`

![selection_020](https://user-images.githubusercontent.com/10103340/39658369-355bc2f0-4fc8-11e8-857c-548c0741a387.png)

**Comment by n3oxmind** [comment]
update:
Tilix 1.7.9 added new token ${activeTerminlTitle}, this will allow a very nice multiple tabs with the correct name see bellow:
![selection_025](https://user-images.githubusercontent.com/10103340/39680686-627ad8de-5158-11e8-94d2-fcdd304d7cd7.png)

here is how to configure Tilix for multiple tabs with gns3
1- Tilix preferences-> make sure to set the highlighted variables as bellow:
![tilix preferences_028](https://user-images.githubusercontent.com/10103340/39680764-1ed58f42-5159-11e8-9654-73ebc0fac4e8.png)
2- gns3 console setting 
```gns3 
tilix -a app-new-session -t "%d" -e "telnet %h %p"
```
all set enjoy nice multiple tabs with Tilix.

**Comment by ahbali** [comment]
@n3oxmind @grossmj : if its a terminal issue why is Dynamips working fine with the default Terminal preferences (Tilix included)?
@n3oxmind : i will use your method when the Tilix PPA gets updated to 1.7.9 ( because it is still  1.7.7 for me),  to work around the bug until it is resolved.

**Comment by n3oxmind** [comment]
@ahbali, 

> using ubuntu 16.04 and gns3 ppa.

have you tried to install gns3 from the source or using pip3,
try this and see if you still have the same issue:
```sh
sudo pip3 install gns3-gui==2.1.5
sudo pip3 install gns3-server==2.1.5
```
if you dont have pip3 installed:
```sh
sudo apt-get update
sudo apt-get install python3-pip
```

**Comment by sliddjur** [comment]
I got tilix working as described. However, is it possible to open the same terminal if I open console multipe times from the same device from within GNS3?
The behaviour now, is that it opens a new console session each time.

**Comment by dhruvrauthan** [comment]
@sliddjur Did you ever find a fix for this? (opening multiple terminals for the same device/appliance)

**Comment by sliddjur** [comment]
> @sliddjur Did you ever find a fix for this? (opening multiple terminals for the same device/appliance)
No, I didn't look into it any more. What I found was that that I use tilix with "quake mode", and that also ruins some of the settings when trying to open a new tab/window in the same tilix sessions, so I pretty much gave up and just use gnome terminal instead.
